modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1900,13 +1900,36 @@ def str_split(self, pat=None, n=-1, expand=False, regex=None):
 
     # END String map partitions operations
 
-    def unique(self):
-        new_modin_frame = self._modin_frame.apply_full_axis(
-            0,
-            lambda x: x.squeeze(axis=1).unique(),
-            new_columns=self.columns,
+    def unique(self, keep="first", ignore_index=True, subset=None):
+        # kernels with 'pandas.Series.unique()' work faster
+        can_use_unique_kernel = (
+            subset is None and ignore_index and len(self.columns) == 1 and keep
         )
-        return self.__constructor__(new_modin_frame)
+
+        if not can_use_unique_kernel and not RangePartitioning.get():
+            return super().unique(keep=keep, ignore_index=ignore_index, subset=subset)
+
+        if RangePartitioning.get():
+            new_modin_frame = self._modin_frame._apply_func_to_range_partitioning(
+                key_columns=self.columns.tolist() if subset is None else subset,
+                func=(
+                    (lambda df: pandas.DataFrame(df.squeeze(axis=1).unique()))
+                    if can_use_unique_kernel
+                    else (
+                        lambda df: df.drop_duplicates(
```

Comment:
Why are we not using this lambda function in the  case where `RangePartitioning` is false?, Is this beneficial only for range partitioning

Reply:
In case of `.drop_duplicates()` it's always beneficial to use an [implementation from `BaseQueryCompiler`](https://github.com/dchigarev/modin/blob/df12976e895f4840c6368e5b3350e1f007dc9fb7/modin/core/storage_formats/base/query_compiler.py#L1790-L1797) instead of calling a full-column kernel in pandas qc. Here are the measurements where I compare BaseQC and PandasQC-full-axis implementations of `.drop_duplicates()`:
![image](https://github.com/modin-project/modin/assets/62142979/4941ded1-86d8-4c56-8134-10d5ef0e4dbc)

Reply:
Got it thanks